great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1360,10 +1360,15 @@ def get_batch_data_and_markers(
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
-        if isinstance(batch_spec, RuntimeQueryBatchSpec):
+        # TODO: clean this up
+        # fix this when building the batch spec or stop doing instance checks at all
+        # and check for the presence of the query key
+        query: Optional[str] = batch_spec.get(
+            "query", getattr(batch_spec, "query", None)
+        )
+        if isinstance(batch_spec, RuntimeQueryBatchSpec) or query:
```

Comment:
I wouldn't be surprised if this was leading to other problems with `QueryAsset` even with `temp_table=True` because of how the `SqlAlchemyBatchData` was being built (not using `query`).